SQL_Queries.txt

```diff
@@ -0,0 +1,1670 @@
++------+----------+---------+
+| 2003 | Liu      | Sanjose |
+| 2004 | Grass    | Berlin  |
+| 2008 | Cisneros | Sanjose |
++------+----------+---------+
+
+23) Select all customers with a rating above 200.
+
+SELECT cnum , cname , city , rating FROM customers WHERE rating > 200 ;
+output:
++------+----------+---------+--------+
+| cnum | cname    | city    | rating |
++------+----------+---------+--------+
+| 2004 | Grass    | Berlin  |    300 |
+| 2008 | Cisneros | Sanjose |    300 |
++------+----------+---------+--------+
+
+24) Count the number of salespeople currently having orders in the orders table. 
+SELECT COUNT(DISTINCT(snum)) AS Salespeople_Orders FROM orders ;
++--------------------+
+| Salespeople_Orders |
++--------------------+
+|                  5 |
++--------------------+
+
+25) Write a query that produces all customers serviced by salespeople with a commission above 12%. Output the customer’s name, salesperson’s name and the salesperson’s rate of commission. 
+
+SELECT c.cname , s.sname , s.comm FROM customers as c
+INNER JOIN
+salespeople AS s WHERE s.snum = c.snum AND s.comm>12 ;
+
+output:
++----------+--------+------+
+| cname    | sname  | comm |
++----------+--------+------+
+| Liu      | Serres |   13 |
+| Grass    | Serres |   13 |
+| Cisneros | RifKin |   15 |
++----------+--------+------+
+
+26) Find salespeople who have multiple customers. 
+
+SELECT s.snum , s.sname , COUNT(c.snum) AS count FROM salespeople AS s
+INNER JOIN
+customers AS c WHERE c.snum = s.snum GROUP BY c.snum HAVING count>1 ;
+output:
++------+--------+-------+
+| snum | sname  | count |
++------+--------+-------+
+| 1001 | Peel   |     2 |
+| 1002 | Serres |     2 |
++------+--------+-------+
+
+27) Find salespeople with customers located in their own cities.
+
+SELECT s.sname , c.cname , s.city FROM salespeople AS s
+INNER JOIN
+customers AS c WHERE c.city = s.city AND c.snum = s.snum ;
+output:
++--------+---------+---------+
+| sname  | cname   | city    |
++--------+---------+---------+
+| Peel   | Hoffman | London  |
+| Serres | Liu     | Sanjose |
+| Peel   | Clemens | London  |
++--------+---------+---------+
+3 rows in set (0.00 sec)
+
+28) Find all salespeople whose name starts with ‘P’ and fourth character is ‘I’.
+SELECT * FROM salespeople WHERE SNAME LIKE "P__I%";
+Empty set (0.03 sec)
+
+29)  Write a query that uses a subquery to obtain all orders for the customer named ‘Cisneros’. Assume you do not know his customer number. 
+
+SELECT onum , amt , odate , cnum FROM orders WHERE cnum IN (SELECT cnum FROM customers WHERE cname="cisneros") ;
+output:
++------+---------+----------+------+
+| onum | amt     | odate    | cnum |
++------+---------+----------+------+
+| 3001 |   18.69 | 10/03/90 | 2008 |
+| 3006 | 1098.16 | 10/03/90 | 2008 |
++------+---------+----------+------+
+2 rows in set (0.00 sec)
+
+30) Find the largest orders for Serres and Rifkin. 
+
+SELECT s.sname , MAX(o.amt) FROM salespeople AS s
+INNER JOIN
```

Comment:
Not used best practice to use joins In Inner Joins you can use 'On' clause or not ?